eno-core/src/main/java/fr/insee/eno/core/processing/out/steps/lunatic/LunaticAddControlFormat.java

```diff
@@ -192,20 +198,28 @@ String formatDateToFrench(String date) {
      * @param format format string
      * @param responseName date picker response attribute
      */
+    private Optional<ControlType> getFormatControlFromDatepickerAttributes(String id, String minValue, String maxValue, String format, String responseName) {
+        if (format == null)
+            throw new RequiredPropertyException("Format is missing in date question '" + id + "'");
 
-    Optional<ControlType> getFormatControlFromDatepickerAttributes(String id, String minValue, String maxValue, String format, String responseName) {
         String controlIdPrefix = id + "-format-date";
 
-        String formattedMinValue = minValue != null ? formatDateToFrench(minValue) : null;
-        String formattedMaxValue = maxValue != null ? formatDateToFrench(maxValue) : null;
-
-        if (minValue == null && maxValue == null) {
-            logger.warn("Aucune contrainte de date définie pour l'id : {}.", id);
-            return Optional.empty();
+        DateFormatter.Result formattedMinValue = validateAndConvertDate(minValue, format);
+        if (formattedMinValue != null && !formattedMinValue.isValid()) {
+            String message = "Invalid value for min date of question '" + id + "': " + formattedMinValue.getErrorMessage();
+            log.error(message);
+            throw new InvalidValueException(message);
+        }
+        DateFormatter.Result formattedMaxValue = validateAndConvertDate(maxValue, format);
+        if (formattedMaxValue != null && !formattedMaxValue.isValid()) {
+            // Due to a bug in the Pogues -> DDI transformation, an invalid max date doesn't throw an exception for now
+            log.warn("Invalid value for max date of question '" + id + "': " + maxValue);
         }
 
-        if (minValue != null && maxValue != null) {
+        boolean generateMin = formattedMinValue != null && formattedMinValue.isValid();
+        boolean generateMax = formattedMaxValue != null && formattedMaxValue.isValid();
 
+        if (generateMin && generateMax) {
             String controlExpression = String.format(
                     "not(not(isnull(%s)) and " +
```

Comment:
🤯

Reply:
je fais une erreur de logique ici ?

Reply:
Aaah oui on a un truc bizarre entre Pogues et Lunatic qui sont pas d'accord sur le sens des contrôles. Du coup Eno rajoute "not(...)" sur les expressions de contrôles quelque part entre Pogues et Lunatic.

Ici on génère des contrôles en dur donc on est pas obligé du not(not(...)), mais David avait sans doute fait ça comme ça pour que ça soit plus "cohérent" visuellement avec les autres contrôles

bref......